packages/client/hooks/useSearchFilter.ts

```diff
@@ -3,16 +3,17 @@ import useForm from '~/hooks/useForm'
 
 // Matches 'items' based on the item's 'getValue' result and the value passed to 'onQueryChange'.
 const useSearchFilter = <T>(items: readonly T[], getValue: (item: T) => string) => {
-  const {fields, onChange} = useForm({
+  const {fields, onChange, setValue} = useForm({
     search: {
       getDefault: () => ''
     }
   })
+  const resetQuery = () => setValue('search', '')
   const {search} = fields
   const {value} = search
   const query = value.toLowerCase()
   const filteredItems = useFilteredItems(query, items, (item) => getValue(item).toLowerCase())
-  return {query: value, filteredItems, onQueryChange: onChange}
+  return {query: value, filteredItems, onQueryChange: onChange, resetQuery}
```

Comment:
We could also make this a more general `setQuery` function, but `resetQuery` seems more applicable for most cases (e.g. "X"s for clearing the search query), even as just a convenience method for a `setQuery` method we might expose in the future.